File: 0006/sum-square-difference.js
const amount = 100;
const numbers = [];

for (let i = 1; i <= amount; i++) {
    numbers.push(i);
}

const sumOfSquares = numbers.map((n) => n * n).reduce((sum, n) => sum + n, 0);
const squareOfSums = Math.pow(numbers.reduce((sum, n) => sum + n, 0), 2);

console.log(squareOfSums - sumOfSquares);

File: 0006/sum-square-difference.go
package main

import "fmt"

func main(){
    const amount = 100
    var sum = amount * ( amount + 1) / 2
    var sumOfSquares = (2 * amount + 1) * (amount + 1) * amount / 6
    fmt.Println(sum * sum - sumOfSquares)
}
